system/lib/wasmfs/proxied_file_backend.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+// This file defines the Proxied File and Proxied Backend of the new file
+// system. Current Status: Work in Progress. See
+// https://github.com/emscripten-core/emscripten/issues/15041.
+
+#include "backend.h"
+#include "file.h"
+#include "thread_utils.h"
+#include "wasmfs.h"
+
+namespace wasmfs {
+
+// This class represents a file that is resident on a background pthread. All
+// file operations are proxied to this dedicated thread.
+class ProxiedFile : public DataFile {
+
+  emscripten::SyncToAsync& proxy;
+  std::shared_ptr<DataFile> backendFile;
```

Comment:
why is this called "backend" file in particular?

Reply:
If it's less verbose, let's use just `basefile`. I just wanted to indicate that this was the associated file that we write to on another dedicated thread.